scraper/spiders/utils.py

```diff
@@ -107,3 +109,8 @@ def strip_accents(string):
         for char in unicodedata.normalize("NFD", string)
         if unicodedata.category(char) != "Mn"
     )
+
+
+def datetime_now_aware() -> datetime:
+    """Create a datetime now aware using local timezone."""
+    return datetime.now(tz=tzlocal.get_localzone())
```

Comment:
Legal esse pacote! Ainda não conhecia. Qual a diferença/vantagem dele em relação a usar:

```
from django.utils import timezone
timezone.localtime()
```

Nesse caso retorna `now()` com a timezone do projeto (`America/Fortaleza`).

Reply:
> Legal esse pacote! Ainda não conhecia. Qual a diferença/vantagem dele em relação a usar:
> 
> ```
> from django.utils import timezone
> timezone.localtime()
> ```
> 
> Nesse caso retorna `now()` com a timezone do projeto (`America/Fortaleza`).

Voce tem razao é possivel usar `django.utils.timezone.localtime()`, specialmente por que ele tem uma implementacao correta do `now()` que retorna um `datetime aware now` quando `settings.USE_TZ=True`:
https://github.com/django/django/blob/464a4c0c59277056b5d3c1132ac1b4c6085aee08/django/utils/timezone.py#L188

Porém ele assume que o Django esteja instalado e depende de ter accesso ao settings do projeto pra funcionar corretamente.

Nao acehai uma boa ideia que o modulo `scraper` dependesse do Django, por isso fiz uma implementacao usando o timezone do sistema via pacote `tzlocal`. Outra opcao seria implemetar como a funcao `now` (que passei o link a acima) e sempre gerar datetime aware usando `utcnow` e passar o tz=UTC.

Nesse caso evitaria a dependencia do pacote `tzlocal`.

Reply:
É, de fato, ficaria acoplado ao Django. Mas já que o Django é uma dependência atual do projeto e os raspadores co-existem com a app Django eu preferia ter acoplado a ele do que a uma nova biblioteca. Além disso, os spiders estão deployados em servidores que tem timezones diferentes da aplicação. Acho que seria melhor controlarmos isso do que pegar o do sistema local, além de evitar mais uma dependência. Daí deixamos pra resolver esse problema no futuro caso resolvamos separar os raspadores do backend. 😅  Faz sentido?

Reply:
> É, de fato, ficaria acoplado ao Django. Mas já que o Django é uma dependência atual do projeto e os raspadores co-existem com a app Django eu preferia ter acoplado a ele do que a uma nova biblioteca. Além disso, os spiders estão deployados em servidores que tem timezones diferentes da aplicação. Acho que seria melhor controlarmos isso do que pegar o do sistema local, além de evitar mais uma dependência. Daí deixamos pra resolver esse problema no futuro caso resolvamos separar os raspadores do backend. sweat_smile Faz sentido?

E possivel implementar da mesma forma que o Django, mas sem criar a dependencia.
Eu sei que o raspador esta no mesmo projeto mas o codigo em si pode ficar independente, entao nao seria essa a melhor opcao?

Quanto a questao de controlar o TIMEZONE usado, na verdade isso nao e necessario pois:
- quando criamos to datetime.utcnow e adicionamos tzinfo=utc se tem a informacao completa
- isso permite ao django transformar para o TIMEZONE do projeto na consulta
- e quando serializado tambem podera ser transformado pelo browser no formato do cliente

Reply:
@anapaulagomes atualizei o PR para evitar a nova dependencia.

Reply:
É verdade, @rudaporto. Obrigada pela explicação. Vamos de Django por enquanto mas com isso em mente.